File: frontend/lib/model/crew.dart
/// Author: Łukasz Piętka (FUT 2025)
class Crew {
  final int id; // non editable
  final String name;
  final String room;
  final String role;
  final String imageUrl;
  final DateTime sobrietyDay; //Editable by superAdmin
  final String description;
  final String email; //non editable
  final bool isSuperAdmin; // editable by superAdmin
  final String crewQuest;
  final String phoneNumber;
  double imageAlignmentY;
  final DateTime? lastModified;

  Crew({
    required this.id,
    required this.name,
    required this.room,
    required this.role,
    required this.imageUrl,
    required this.sobrietyDay,
    required this.description,
    required this.crewQuest,
    required this.phoneNumber,
    required this.email,
    required this.isSuperAdmin,
    required this.imageAlignmentY,
    this.lastModified,
  });
}

File: backend/src/main/java/me/havooc/poligon/model/Crew.java
/**
 * @Author Łukasz Piętka (FUT 2025)
 */
package me.havooc.poligon.model;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.UpdateTimestamp;

import java.sql.Timestamp;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Crew {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private String room;
    private String role;
    private String imageUrl;
    private String email;
    private LocalDate sobrietyDay;
    @Column(length = 4096)
    private String description;
    @Column(length = 2048)
    private String crewQuest;
    private String phoneNumber;
    private boolean isSuperAdmin;
    @UpdateTimestamp
    private LocalDateTime lastModified;
    private Double imageAlignmentY;
}